modules/gapi/src/backends/plaidml/gplaidmlbackend.cpp

```diff
@@ -0,0 +1,286 @@
+
+#include <opencv2/gapi/gcommon.hpp>
+#include <opencv2/gapi/util/any.hpp>
+#include <opencv2/gapi/gtype_traits.hpp>
+
+#include "compiler/gobjref.hpp"
+#include "compiler/gmodel.hpp"
+
+#include "backends/plaidml/gplaidmlbackend.hpp"
+#include "backends/plaidml/plaidml_util.hpp"
+
+#include "api/gbackend_priv.hpp" // FIXME: Make it part of Backend SDK!
+
+using GPlaidMLModel = ade::TypedGraph
+    < cv::gimpl::PlaidMLUnit
+    , cv::gimpl::Protocol
+    >;
+
+// FIXME: Same issue with Typed and ConstTyped
+using GConstGPlaidMLModel = ade::ConstTypedGraph
+    < cv::gimpl::PlaidMLUnit
+    , cv::gimpl::Protocol
+    >;
+
+namespace
+{
+    class GPlaidMLBackendImpl final: public cv::gapi::GBackend::Priv
+    {
+        virtual void unpackKernel(ade::Graph            &graph,
+                                  const ade::NodeHandle &op_node,
+                                  const cv::GKernelImpl &impl) override
+        {
+            GPlaidMLModel gm(graph);
+            auto plaidml_impl = cv::util::any_cast<cv::GPlaidMLKernel>(impl.opaque);
+            gm.metadata(op_node).set(cv::gimpl::PlaidMLUnit{plaidml_impl});
+        }
+
+        virtual EPtr compile(const ade::Graph &graph,
+                             const cv::GCompileArgs &,
+                             const std::vector<ade::NodeHandle> &nodes) const override
+        {
+            return EPtr{new cv::gimpl::GPlaidMLExecutable(graph, nodes)};
+        }
+   };
+}
+
+cv::gapi::GBackend cv::gapi::plaidml::backend()
+{
+    static cv::gapi::GBackend this_backend(std::make_shared<GPlaidMLBackendImpl>());
+    return this_backend;
+}
+
+void cv::gimpl::GPlaidMLExecutable::initBuffers(const std::vector<cv::gimpl::Data>& data,
+                                                std::vector<plaidml::exec::Binding>& bindings)
+{
+
+    // NB: This is necessary because we keep a pointer to bindings elements to buffer_map
+    // In order to them to remain valid it's required to prevant reallocation
+    bindings.reserve(data.size());
+    for (const auto& d : data)
+    {
+        GAPI_Assert(d.meta.index() == cv::GMetaArg::index_of<cv::GMatDesc>() &&
+                    "Now PlaidML backend supported only cv::GMat's");
+
+        const auto& desc = cv::util::get<cv::GMatDesc>(d.meta);
+
+        auto placeholder = plaidml::edsl::Placeholder(
+                           cv::util::plaidml::depth_from_ocv(desc.depth),
+                           {desc.size.width, desc.size.height, desc.chan});
+
+        const auto& shape = placeholder.shape();
+        plaidml::TensorShape tshape(shape.dtype(), shape.int_dims());
+        plaidml::Buffer buffer(device_id_, tshape);
+
+        bindings.push_back(plaidml::exec::Binding{std::move(placeholder),
+                                                  std::move(buffer)});
+
+        auto& tensor_map = m_res.slot<plaidml::edsl::Tensor>();
+        // FIXME Avoid Copy here !!!
+        tensor_map.emplace(d.rc, placeholder);
```

Comment:
If your `placeholder` really `move`s two lines above, you store here nothing.

Does `Binding` take `placeholder`s by value? If it takes a reference, I'd construct an object in the map first and then push it to bindings (referred to via reference).

Reply:
Fixed